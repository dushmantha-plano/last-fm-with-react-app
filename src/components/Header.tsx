import * as React from 'react';
import { NavLink } from 'react-router-dom';
import SidePanel from './SidePanel/SidePanel';
import styled from 'styled-components';

interface HeaderProps {}

const Header: React.FunctionComponent<HeaderProps> = (): React.ReactElement => (
  <Wrapper>
    <HeaderInner>
      <HeaderNav>
        <HeaderNavLink exact={true} to="/albums" activeClassName="is-active">
          Albums
        </HeaderNavLink>
        <HeaderNavLink exact={true} to="/tracks" activeClassName="is-active">
          Tracks
        </HeaderNavLink>
      </HeaderNav>
    </HeaderInner>
    <SidePanel />
  </Wrapper>
);

export default Header;

// Styled components

const Wrapper = styled('header')`
  align-items: center;
  background: black;
  color: white;
  display: flex;
  font-family: 'Poppins', sans-serif;
  height: 80px;
  justify-content: space-between;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 99;
  @media (min-width: 320px) and (max-width: 768px) {
    flex-direction: column;
  }
`;

const HeaderInner = styled('div')`
  align-items: center;
  display: flex;
  flex-direction: row;
  height: 100%;
  justify-content: flex-start;
  padding: 0 10px;
`;

const HeaderNav = styled('nav')`
  align-items: center;
  display: flex;
  @media (min-width: 992px) {
    margin: 0;
  }
`;

const HeaderNavLink = styled(NavLink)`
  align-items: center;
  color: #dcdcdc;
  color: #dcdcdc;
  display: flex;
  margin: 0 25px;
  :link {
    text-decoration: none;
  }

  :visited {
    text-decoration: none;
  }

  :hover {
    color: white;
    text-decoration: none;
  }

  :active {
    text-decoration: none;
  }
`;
