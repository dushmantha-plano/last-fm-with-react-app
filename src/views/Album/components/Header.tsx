import * as React from 'react';
import styled from 'styled-components';

export interface HeaderProps {
  image: string;
  artist: string;
  name: string;
  playCount: number;
}

const Header: React.FunctionComponent<HeaderProps> = ({
  name,
  image,
  artist,
  playCount,
}: HeaderProps): React.ReactElement => (
  <React.Fragment>
    <BackgroundContainer>
      <ImageWrapper>
        <AlbumImage
          style={{
            backgroundImage: 'url(' + image + ')',
          }}
        />
      </ImageWrapper>
      <ImageOverlay />
    </BackgroundContainer>
    <DataContainer>
      <AlbumName>{name}</AlbumName>
      <AlbumData>
        <HeaderContainer>
          <HeaderTitle>Artist</HeaderTitle>
          <AlbumArtistValue>{artist}</AlbumArtistValue>
        </HeaderContainer>
        <AlbumPlayCountContainer>
          <AlbumPlayCountTitle>Times played</AlbumPlayCountTitle>
          <AlbumPlayCountValue>
            {playCount.toLocaleString()}
          </AlbumPlayCountValue>
        </AlbumPlayCountContainer>
      </AlbumData>
    </DataContainer>
  </React.Fragment>
);

export default Header;

// Styled components

const BackgroundContainer = styled('div')`
  background-color: #161b1e;
  height: 100%;
  position: absolute;
  width: 100%;
`;

const ImageWrapper = styled('div')`
  height: 100%;
  position: absolute;
  right: 0%;
  width: 50%;
`;

const AlbumImage = styled('img')`
  background-position: 50% 25%;
  background-repeat: no-repeat;
  background-size: cover;
  height: 100%;
  width: 100%;
`;

const ImageOverlay = styled('div')`
  background: linear-gradient(0.25turn, rgb(22, 27, 30, 1), rgb(22, 27, 30, 0));
  height: 100%;
  position: absolute;
  right: 0;
  width: 50%;
`;

const DataContainer = styled('div')`
  padding: 30px 20px 20px 30px;
  position: relative;
  z-index: 1;
`;

const AlbumName = styled('p')`
  font-size: 40px;

  @media (min-width: 320px) and (max-width: 768px) {
    font-size: 30px;
  }
`;

const AlbumData = styled('div')``;

const HeaderContainer = styled('div')`
  align-items: baseline;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  margin-top: 20px;
  width: 80px;
`;
const HeaderTitle = styled('p')`
  font-size: 10px;
`;
const AlbumArtistValue = styled('p')`
  color: white;
  font-size: 20px;
  width: max-content;
`;

const AlbumPlayCountContainer = styled('div')`
  align-items: baseline;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  margin-top: 20px;
  width: 80px;
`;
const AlbumPlayCountTitle = styled('p')`
  font-size: 10px;
`;
const AlbumPlayCountValue = styled('p')`
  color: white;
  font-size: 20px;
`;
