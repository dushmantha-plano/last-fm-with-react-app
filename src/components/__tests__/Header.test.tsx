import 'jest-styled-components';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import Header from '../Header';
import React from 'react';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
describe('Test for Header Component', () => {
  test('Should render the component', () => {
    const wrapper = shallow(<Header />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
