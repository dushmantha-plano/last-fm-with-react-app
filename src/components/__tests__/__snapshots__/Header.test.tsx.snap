// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Header Component Should render the component 1`] = `
<styled.header>
  <styled.div>
    <styled.nav>
      <Styled(NavLink)
        activeClassName="is-active"
        exact={true}
        to="/albums"
      >
        Albums
      </Styled(NavLink)>
      <Styled(NavLink)
        activeClassName="is-active"
        exact={true}
        to="/tracks"
      >
        Tracks
      </Styled(NavLink)>
    </styled.nav>
  </styled.div>
  <SidePanel />
</styled.header>
`;
